src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -365,6 +408,43 @@ public IRecordingDescriptor startRecording(
 
         return builder.build();
     }
+
+    private void scheduleRecordingStopNotification(
+            String recordingName, long delay, ConnectionDescriptor connectionDescriptor) {
+        ScheduledFuture<Optional<IRecordingDescriptor>> scheduledFuture =
+                this.scheduler.schedule(
+                        () -> {
+                            return targetConnectionManager.executeConnectedTask(
+                                    connectionDescriptor,
+                                    connection -> {
+                                        Optional<IRecordingDescriptor> desc =
+                                                getDescriptorByName(connection, recordingName);
+
+                                        boolean recordingStopped =
+                                                desc.isPresent()
+                                                        && !desc.stream()
+                                                                .filter(
+                                                                        recording ->
+                                                                                recording.getState()
+                                                                                        == RecordingState
+                                                                                                .STOPPED)
```

Comment:
I think this could be tidied up just slightly:

```java
desc.stream()
  .map(IRecordingDescriptor::getState)
  .filter(s -> s.equals(RecordingState.STOPPED))
  .count()
```